source/extensions/filters/http/aws_lambda/aws_lambda_filter.cc

```diff
@@ -157,7 +157,8 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
   }
 
   if (payload_passthrough_) {
-    setLambdaHeaders(headers, arn_, invocation_mode_);
+    headers.setHost(std::format("https://lambda.{}.amazonaws.com", arn_->region()))
```

Comment:
The lambda endpoints value may vary. https://docs.aws.amazon.com/general/latest/gr/lambda-service.html

Can we just set this host header inside `setLambdaHeaders` function in L44?
```
headers.setHost("lambda");
```
this should work as well.

Reply:
this has been updated.  If you can point me to where this extension derives which endpoint it should hit I can also just set the header to that

Reply:
After a quick review it seems that envoy only currently knows of `lambda.REGION.amazonaws.com` from what I can tell

Reply:
The extension just sends it to lambda cluster which will be pointing to a specific endpoint.

From the examples available on Envoy aws lambda extension [documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/aws_lambda_filter) it is only pointing to `lambda.REGION.amazonaws.com` but user can configure any lambda endpoint.

Reply:
In that case should we not just rewrite the host header to what's configured in the cluster then? That seems like the simplest most pragmatic solution.

Rewriting it to "lambda" or anything other than the target seems to result in a failure response from AWS